sw/host/opentitanlib/src/util/parse_int.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::num::ParseIntError;
+
+/// Trait for parsing integers.
+///
+/// Strings beginning with the common and classic prefixes `0x`, `0o`, `0b` or `0` are parsed
+/// as hex, octal, binary, and octal (classic).  Anything else is parsed as decimal.
+/// A leading `+` or `-` is permitted.  Any string parsed by `strtol(3)` or `strtoul(3)`
+/// will be parsed successfully.
+///
+/// This code was shamelessly adapted from https://github.com/Rust-for-Linux/linux/pull/87
+pub trait ParseInt: Sized {
+    fn from_str_radix(src: &str, radix: u32) -> Result<Self, ParseIntError>;
+
+    fn from_str(src: &str) -> Result<Self, ParseIntError> {
```

Comment:
Do you think there's a nicer name like e.g. `from_prefixed_str`? I don't care enough I think.

Reply:
Heh.  No, I think "from_str" is a nice enough name (and the prefixes are optional - you're free to pass in a base-10 representation if you like).